File: backend/models/PhieuNhapHang.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const PhieuNhapHang = sequelize.define(
  "PhieuNhapHang",
  {
    id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true },
    ma_phieu: { type: DataTypes.STRING(20), allowNull: false, unique: true },
    nha_cc_id: { type: DataTypes.BIGINT, allowNull: true },
    nguoi_tao: { type: DataTypes.BIGINT, allowNull: true },
    trang_thai: { type: DataTypes.STRING(20), defaultValue: "draft" }, // draft | completed | cancelled
    giam_gia: { type: DataTypes.DECIMAL(15, 2), defaultValue: 0 },
    tong_tien: { type: DataTypes.DECIMAL(15, 2), defaultValue: 0 },
    ghi_chu: { type: DataTypes.TEXT, allowNull: true },
  },
  {
    tableName: "PhieuNhapHang",
    timestamps: true,
    createdAt: "created_at",
    updatedAt: "updated_at",
  }
);

module.exports = PhieuNhapHang;

File: backend-csharp/Models/PhieuNhapHang.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class PhieuNhapHang
{
    public long Id { get; set; }

    public string MaPhieu { get; set; } = null!;

    public long? NhaCcId { get; set; }

    public long? NguoiTao { get; set; }

    public string? TrangThai { get; set; }

    public decimal? GiamGia { get; set; }

    public decimal? TongTien { get; set; }

    public string? GhiChu { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }

    public virtual ICollection<ChiTietPhieuNhap> ChiTietPhieuNhaps { get; set; } = new List<ChiTietPhieuNhap>();

    public virtual TaiKhoan? NguoiTaoNavigation { get; set; }

    public virtual NhaCungCap? NhaCc { get; set; }
}
